lib/util.js

```diff
@@ -38,15 +38,15 @@ const inspectDefaultOptions = Object.seal({
   breakLength: 60
 });
 
-const CIRCULAR_ERROR_MESSAGE = 'Converting circular structure to JSON';
+const CIRCULAR_ERROR_MESSAGE = /Circular/i;
 
 var Debug;
 
 function tryStringify(arg) {
   try {
     return JSON.stringify(arg);
   } catch (err) {
-    if (err.name === 'TypeError' && err.message === CIRCULAR_ERROR_MESSAGE)
+    if (err.name === 'TypeError' && err.message.match(CIRCULAR_ERROR_MESSAGE))
```

Comment:
Okay. I understand the concerns. How about solving this problem broadly by having engine *specific string templates for various error messages? 
```js
// v8-error-msg.js
var errorMsg = [];
...
errorMsg[engineName]['SyntaxError_Unexpected_Id'] = "SyntaxError: Unexpected identifier";
...
```
We could update test infrastructure to fetch the appropriate error message based on template. Of-course some error message might need parameters like variable names, function names, etc. which we will have to add support for.

```js
// test.js
assert.strictEquals(stdErr, common.engineSpecific('SyntaxError_Unexpected_Id'));
```

Currently the way it is achieved in `node-chakracore` is

```js
assert.strictEquals(stdErr, common.engineSpecific({
    v8: "SyntaxError: Unexpected identifier",
    chakracore: "SyntaxError: Expected ';'"})
}));
```
Additional benefit we get is if engine changes the error message, there will be one common place where the messages will have to be changed. I know this is non-trivial work since it will involve going through every test case that relies on error message, make the entry of error message in central engine specific file and then update the test to fetch that error message, but wanted to get thoughts of community to make node test cases more engine neutral.

cc: @addaleax, @jasnell 

*Roughly adopted from https://github.com/nodejs/node/pull/4311 .